drivers/spi/spi_silabs_eusart.c

```diff
@@ -37,11 +63,31 @@ struct spi_silabs_eusart_config {
 	const struct silabs_clock_control_cmu_config clock_cfg;
 	uint32_t clock_frequency;
 	const struct pinctrl_dev_config *pcfg;
+	volatile uint8_t mosi_overrun;
 };
 
+/* Global variable */
+#ifdef CONFIG_SPI_SILABS_EUSART_DMA
+static volatile uint8_t empty_buffer;
+#endif
+
+/* Function declaration */
+#ifdef CONFIG_SPI_SILABS_EUSART_DMA
+void spi_silabs_dma_rx_callback(const struct device *dev, void *user_data, uint32_t channel,
+				int status);
+#endif
+
 /* Helper Functions */
-static int spi_silabs_eusart_configure(const struct device *dev, const struct spi_config *config,
-			     uint16_t *control)
+#ifdef CONFIG_SPI_SILABS_EUSART_DMA
+static inline bool is_dma_enabled_instance(const struct device *dev)
+{
+	struct spi_silabs_eusart_data *data = dev->data;
+
+	return data->dma_channel_rx.dma_dev != NULL;
+}
+#endif
```

Comment:
I suggest to avoid conditional declaration of a function (note I am not talking about the definition). The caller will have to place the call under `#ifdef`. I suggest this alternative:
```
static inline bool is_dma_enabled_instance(const struct device *dev)
{
#ifdef CONFIG_SPI_SILABS_EUSART_DMA
	struct spi_silabs_eusart_data *data = dev->data;

	return data->dma_channel_rx.dma_dev != NULL;
#else
	return false;
#endif
}
```

This should allow to simplify some lines of code.